subprojects/execution/src/main/java/org/gradle/internal/execution/history/changes/AbstractFingerprintChanges.java

```diff
@@ -14,38 +14,27 @@
  * limitations under the License.
  */
 
-package org.gradle.api.internal.changedetection.rules;
+package org.gradle.internal.execution.history.changes;
 
 import com.google.common.collect.ImmutableSortedMap;
-import org.gradle.api.NonNullApi;
-import org.gradle.api.internal.changedetection.state.TaskExecution;
-import org.gradle.internal.changes.TaskStateChangeVisitor;
+import org.gradle.internal.change.ChangeContainer;
+import org.gradle.internal.change.ChangeVisitor;
+import org.gradle.internal.fingerprint.CurrentFileCollectionFingerprint;
 import org.gradle.internal.fingerprint.FileCollectionFingerprint;
 
-@NonNullApi
-public abstract class AbstractNamedFileSnapshotTaskStateChanges implements TaskStateChanges {
-    protected final TaskExecution previous;
-    protected final TaskExecution current;
+public abstract class AbstractFingerprintChanges implements ChangeContainer {
+    protected final ImmutableSortedMap<String, ? extends FileCollectionFingerprint> previous;
```

Comment:
The `? extends` is not necessary anymore for `previous` and `current`.

Reply:
Good point, fixed.